scripts/Diffraction/single_crystal/base_sx.py

```diff
@@ -614,6 +618,155 @@ def remove_non_integrated_peaks(peaks):
+                            )  # outer radius
+                            ax.add_patch(patch)
+                            patch = Circle(
+                                (0, 0), bg_inner_radii[imax] / box_lengths[imax], facecolor="none", edgecolor=3 * [0.7], ls="--"
+                            )  # inner radius
+                            ax.add_patch(patch)
+                        # format axes
+                        ax.set_xlim(-1, 1)
+                        ax.set_ylim(-1, 1)
+                        ax.set_aspect("equal")
+                        ax.set_xlabel(ws_cut.getDimension(dims[0]).name)
+                        ax.set_ylabel(ws_cut.getDimension(dims[1]).name)
+                    fig.suptitle(
+                        f"{ipk} ({','.join(str(np.round(pk.getHKL(), 2))[1:-1].split())})"
+                        f"  $I/\\sigma$={np.round(pk.getIntensityOverSigma(), 2)}\n"
+                        rf"$\lambda$={np.round(pk.getWavelength(), 2)} $\AA$; "
+                        rf"$2\theta={np.round(np.degrees(pk.getScattering()), 1)}^\circ$; "
+                        rf"d={np.round(pk.getDSpacing(), 2)} $\AA$"
+                    )
+                    fig.tight_layout()
+                    pdf.savefig(fig)
+                    close(fig)
+
+                if ws_cut is not None:
+                    mantid.DeleteWorkspace(ws_cut)
+        except OSError:
+            raise RuntimeError(
+                f"OutputFile ({filename}) could not be opened - please check it is not open by "
+                f"another programme and that the user has permission to write to that directory."
+            )
+
+    @staticmethod
+    def _bin_MD_around_peak(wsMD, pk, peak_shape, nbins_max, extent, frame_to_peak_centre_attr):
+        shape_info = json_loads(peak_shape.toJSON())
+        if peak_shape.shapeName().lower() == "spherical":
+            BaseSX.convert_spherical_representation_to_ellipsoid(shape_info)
+        # get radii
+        radii = np.array([shape_info[f"radius{iax}"] for iax in range(3)])
+        bg_inner_radii = np.array([shape_info[f"background_inner_radius{iax}"] for iax in range(3)])
+        bg_outer_radii = np.array([shape_info[f"background_outer_radius{iax}"] for iax in range(3)])
+        isort = np.argsort(radii)
+        imax = isort[-1]
+        # eignevectors of ellipsoid
+        evecs = np.zeros((3, 3))
+        for iax in range(len(radii)):
+            evec = np.array([float(elem) for elem in shape_info[f"direction{iax}"].split()])
+            evecs[:, iax] = evec / np.linalg.norm(evec)
+        # center and translation
+        translation = np.array([shape_info[f"translation{iax}"] for iax in range(3)])
+        cen = getattr(pk, frame_to_peak_centre_attr)()
+        cen = np.matmul(evecs.T, np.array(cen) + translation)
+        # get extents
+        box_lengths = bg_outer_radii if np.all(bg_outer_radii > 1e-10) else radii
```

Comment:
This tolerance seems to be used elsewhere in the file, can it also be a constant at the top of the file?

Reply:
It's just my default tolerance for float comparison for float comparisons, I can add it to a variable as well but it is arbitrary! Do you know a better way to do this? Using `np.isclose` with default tolerances?

Reply:
I think you're right you could use `np.isclose`! As long as you dont mind its default tolerance is `1e-08`

Reply:
I ended up using `np.allclose` - about to push the suggested changes!